src/auth/externalclient.ts

```diff
@@ -64,6 +70,18 @@ export class ExternalAccountClient {
     options: ExternalAccountClientOptions,
     additionalOptions?: RefreshOptions
   ): BaseExternalAccountClient | null {
+    const workforceAudiencePattern = new RegExp(WORKFORCE_AUDIENCE_PATTERN);
+    if (
+      options &&
+      options.workforce_pool_user_project &&
+      options.audience &&
+      !options.audience.match(workforceAudiencePattern)
+    ) {
+      throw new Error(
+        'The workforce_pool_user_project parameter should only be provided ' +
+          'for a Workforce Pool configuration.'
+      );
+    }
```

Comment:
This check should be done in the IdentityPoolClient constructor (otherwise they can initialize it themselves and we wouldn't catch this error).